esp-riscv-rt/src/lib.rs

```diff
@@ -829,36 +829,39 @@ _vector_table:
 r#"
 #this is required for the linking step, these symbols for in-use interrupts should always be overwritten by the user.
 .section .trap, "ax"
-.weak cpu_int_1_handler
-.weak cpu_int_2_handler
-.weak cpu_int_3_handler
-.weak cpu_int_4_handler
-.weak cpu_int_5_handler
-.weak cpu_int_6_handler
-.weak cpu_int_7_handler
-.weak cpu_int_8_handler
-.weak cpu_int_9_handler
-.weak cpu_int_10_handler
-.weak cpu_int_11_handler
-.weak cpu_int_12_handler
-.weak cpu_int_13_handler
-.weak cpu_int_14_handler
-.weak cpu_int_15_handler
-.weak cpu_int_16_handler
-.weak cpu_int_17_handler
-.weak cpu_int_18_handler
-.weak cpu_int_19_handler
-.weak cpu_int_20_handler
-.weak cpu_int_21_handler
-.weak cpu_int_22_handler
-.weak cpu_int_23_handler
-.weak cpu_int_24_handler
-.weak cpu_int_25_handler
-.weak cpu_int_26_handler
-.weak cpu_int_27_handler
-.weak cpu_int_28_handler
-.weak cpu_int_29_handler
-.weak cpu_int_30_handler
-.weak cpu_int_31_handler
+// See https://github.com/esp-rs/esp-hal/issues/1326 and https://reviews.llvm.org/D98762
+// and yes, this all has to go on one line... *sigh*.
+.lto_discard interrupt1, interrupt2, interrupt3, interrupt4, interrupt5, interrupt6, interrupt7, interrupt8, interrupt9, interrupt10, interrupt11, interrupt12, interrupt13, interrupt14, interrupt15, interrupt16, interrupt17, interrupt18, interrupt19, interrupt20, interrupt21, interrupt22, interrupt23, interrupt24, interrupt25, interrupt26, interrupt27, interrupt28, interrupt29, interrupt30, interrupt31
```

Comment:
decided to rename them all `interruptX` to be consistent with xtensa, and also because this line was massive (still is but :shrug:)